### PR TITLE
feat: Delete VASP Button in Basic Details Menu

### PR DESCRIPTION
**Pull request by elysee15**
SC-2871

### CODE REVIEW

File: web/gds-admin-ui/src/__tests__/basic-details.spec.js [hunk 1]
```diff
@@ -3,252 +3,158 @@ import userEvent from '@testing-library/user-event'
...
-                    "pid": "0",
-                    "version": "4"
+    describe("ReviewOption", () => {
+        it("Should not be disable when status is different to pending", async () => {
+
+            const mockVaspData = {
+                "vasp": {
+                    "verification_status": "PENDING_REVIEW",
+                }
+            }
```
Comment (pdeziel) [inline web/gds-admin-ui/src/__tests__/basic-details.spec.js:19]: In the backend we ended up allowing registrations in states <= `PENDING` or in the `ERRORED` state to be deleted. We should make the same change in the UI. In summary, we want to allow deletion in states: `NO_VERIFICATION`, `SUBMITTED`, `EMAIL_VERIFIED`, `PENDING_REVIEW`, and `ERRORED`

and prevent deletion in states: `REVIEWED`, `ISSUING_CERTIFICATE`, `VERIFIED`, `REJECTED`, and `APPEALED`.

File: web/gds-admin-ui/src/components/DeleteModal.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import ButtonSpinner from './ButtonSpinner';
+import warningImg from 'assets/images/warning.svg'
+import { ModalCloseButton } from './Modal';
+import PropTypes from 'prop-types'
+
+function DeleteModal({ onDelete, isLoading = false }) {
+
+    return <div className='text-center'>
+        <img src={warningImg} alt="Warning" />
+        <h4>Are you sure?</h4>
```
Comment (pdeziel) [inline web/gds-admin-ui/src/components/DeleteModal.js:12]: Can we change the heading message to something like the following?

`Are you sure you want to delete registration <registration_name>?`

We want to make it as clear as possible to the user what they are deleting.

File: web/gds-admin-ui/src/pages/app/details/BasicDetails/components/DeleteVaspModal.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        if (params && params.id) {
+
+            deleteVasp(params.id).then(() => {
+                setIsLoading(false)
+                history.replace('/vasps')
+
+
+                toast.success("Registration deleted successfully", { duration: 7000 })
+            }).catch(error => {
+                toast.error("You are not able to delete this registration")
```
Comment (pdeziel) [inline web/gds-admin-ui/src/pages/app/details/BasicDetails/components/DeleteVaspModal.js:24]:
```suggestion
                toast.error("Unable to delete registration")
```

File: web/gds-admin-ui/src/utils/index.js [hunk 4]
```diff
@@ -236,8 +236,8 @@ const getConductsCustomerKYC = (status) => status ? "does" : "does not"
 const getMustSafeguardPii = (status) => status ? "must" : "is not required to"
 const getSafeguardPii = (status) => status ? "does" : "does not"
 
-function isEditMenuAvailable(verificationStatus) {
+function isOptionAvailable(verificationStatus) {
     return ["NO_VERIFICATION", "SUBMITTED", "EMAIL_VERIFIED", "PENDING_REVIEW"].includes(verificationStatus)
```
Comment (pdeziel) [inline web/gds-admin-ui/src/utils/index.js:240]:
```suggestion
    return ["NO_VERIFICATION", "SUBMITTED", "EMAIL_VERIFIED", "PENDING_REVIEW", "ERRORED"].includes(verificationStatus)
```

Comment (pdeziel) [reply web/gds-admin-ui/src/utils/index.js:240]: We should add `ERRORED` to this list per my comment above`.

**Review by pdeziel — CHANGES_REQUESTED** [review]
This is looking good, thanks for adding the tests and the logic for when the delete button should be enabled. I just have a few requested changes below before we merge this in.

File: web/gds-admin-ui/src/components/DeleteModal.js (hunk 2 above)
Comment (elysee15) [reply web/gds-admin-ui/src/components/DeleteModal.js:12]: you right

**Review by pdeziel — APPROVED** [review]
Looks good to me, thanks for making the changes!

**Pull request merged**